### PR TITLE
Pass input artifact types to kubeflow container entrypoint

### PR DESCRIPTION
**Pull request by schustmi**
## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Describe changes
- Use correct input artifact type in kubeflow containers

### CODE REVIEW

**Review by AlexejPenner — COMMENTED** [review]
Looks good :)

File: src/zenml/integrations/kubeflow/orchestrators/kubeflow_component.py [hunk 1]
```diff
@@ -100,6 +122,8 @@ def __init__(
             step_module,
             "--step_function_name",
             step_function_name,
+            "--input_artifact_types",
+            json.dumps(input_artifact_type_mapping),
```
Comment (htahir1) [inline src/zenml/integrations/kubeflow/orchestrators/kubeflow_component.py:126]: Could there be potential exceptions here?

File: src/zenml/integrations/kubeflow/container_entrypoint.py [hunk 2]
```diff
@@ -481,6 +493,7 @@ def main() -> None:
             step_source_module_name=args.step_module,
             step_function_name=args.step_function_name,
             executor_class_target_module_name=executor_class_target_module_name,
+            input_artifact_type_mapping=json.loads(args.input_artifact_types),
```
Comment (htahir1) [inline src/zenml/integrations/kubeflow/container_entrypoint.py:496]: Could there be potential exceptions here?

File: src/zenml/integrations/kubeflow/container_entrypoint.py [hunk 3]
```diff
@@ -385,8 +390,14 @@ def _create_executor_class(
     materializers = step_instance.get_materializers(ensure_complete=True)
 
     input_spec = {}
-    for key, value in step_class.INPUT_SIGNATURE.items():
-        input_spec[key] = type_registry.get_artifact_type(value)[0]
+    for input_name, class_path in input_artifact_type_mapping.items():
+        artifact_class = source_utils.load_source_path_class(class_path)
+        if not issubclass(artifact_class, BaseArtifact):
+            raise RuntimeError(
+                f"Class `{artifact_class}` specified as artifact class for "
```
Comment (htahir1) [inline src/zenml/integrations/kubeflow/container_entrypoint.py:397]: Good one!

**Review by htahir1 — COMMENTED** [review]
Nice! Sorry for the late review. I just added here two notes on whether we should have exception handling but feel free to ignore if you think that is not needed

**Pull request merged**